vllm/v1/core/scheduler.py

```diff
@@ -621,6 +663,8 @@ class SchedulerOutput:
     num_scheduled_tokens: Dict[str, int]
     total_num_scheduled_tokens: int
     scheduled_encoder_inputs: Dict[str, List[int]]
+    use_spec_decode: bool
+    scheduled_spec_decode_tokens: Dict[str, List[int]]
```

Comment:
Maybe a more general approach:
1. no specific to spec decode: add a function to scheduler so it can schedule n new tokens (with token ids)
2. specific to spec decode: rejection sampling, some tokens might be rejected, some ways of rewinding